waspc/src/Wasp/Generator/ServerGenerator.hs

```diff
@@ -86,11 +87,17 @@ genDotEnv spec = return $
   case AS.dotEnvServerFile spec of
     Just srcFilePath
       | not $ AS.isBuild spec ->
-          [ createCopyFileDraft
+          [ createCopyAndModifyTextFileDraft
               (C.serverRootDirInProjectRootDir </> dotEnvInServerRootDir)
               srcFilePath
+              addDevDbUrlToStartIfSet
           ]
     _ -> []
+  where
+    addDevDbUrlToStartIfSet :: T.Text -> T.Text
+    addDevDbUrlToStartIfSet t = case AS.devDatabaseUrl spec of
+      Nothing -> t
+      Just url -> "DATABASE_URL=" <> T.pack url <> "\n" <> t
```

Comment:
I'm wondering what would happen if:
- the file did not exist, would we still start the server ok?
- if the file did exist, and already had that envar set?

Reply:
Fixed!